app/src/main/java/com/bottlerocketstudios/brarchitecture/ui/projects/ProjectsViewModel.kt

```diff
@@ -0,0 +1,41 @@
+package com.bottlerocketstudios.brarchitecture.ui.projects
+
+import com.bottlerocketstudios.compose.projects.ProjectsItemState
+
+import com.bottlerocketstudios.brarchitecture.domain.repositories.BitbucketRepository
+import com.bottlerocketstudios.brarchitecture.ui.BaseViewModel
+import com.bottlerocketstudios.compose.util.asMutableState
+import com.bottlerocketstudios.compose.util.formattedUpdateTime
+import kotlinx.coroutines.flow.map
+import org.koin.core.component.inject
+import java.time.Clock
+
+class ProjectsViewModel : BaseViewModel() {
+
+    // DI
+    private val repo: BitbucketRepository by inject()
+    private val clock by inject<Clock>()
+
+    val projectsList = repo.projects.map {
+        it.map { dto ->
+            ProjectsItemState(
+                name = dto.name.asMutableState(),
+                key = dto.key.asMutableState(),
+                updated = dto.updatedOn.formattedUpdateTime(clock).getString().asMutableState()
+            )
+        }
+    }
+
+    // Init logic
+    init {
+        launchIO {
+            getProjects()
```

Comment:
Hey @LuisRocketRangel , could probably just inline this function call in the init block here where you could get rid of that getProjects function below. Would just look something like:

```
init {
        launchIO {
            repo.getProjects()
        }
    }
```